src/OrbitQt/orbitmainwindow.cpp

```diff
@@ -854,16 +854,38 @@ void OrbitMainWindow::on_actionSave_Preset_As_triggered() {
 void OrbitMainWindow::on_actionToggle_Capture_triggered() { app_->ToggleCapture(); }
 
 const QString OrbitMainWindow::kCollectThreadStatesSettingKey{"CollectThreadStates"};
+const QString OrbitMainWindow::kCollectMemoryInfoSettingKey{"CollectMemoryInfo"};
+const QString OrbitMainWindow::kMemorySamplingPeriodNsSettingKey{"MemorySamplingPeriodNs"};
+const QString OrbitMainWindow::kMemoryWarningThresholdKbSettingKey{"MemoryWarningThresholdKb"};
 const QString OrbitMainWindow::kLimitLocalMarkerDepthPerCommandBufferSettingsKey{
     "LimitLocalMarkerDepthPerCommandBuffer"};
 const QString OrbitMainWindow::kMaxLocalMarkerDepthPerCommandBufferSettingsKey{
     "MaxLocalMarkerDepthPerCommandBuffer"};
-;
+
+constexpr uint64_t kMemorySamplingPeriodNsDefaultValue = 100'000'000;        // 100ms
+constexpr uint64_t kMemoryWarningThresholdKbDefaultValue = 1024 * 1024 * 8;  // 8Gb
 
 void OrbitMainWindow::LoadCaptureOptionsIntoApp() {
   QSettings settings;
   app_->SetCollectThreadStates(settings.value(kCollectThreadStatesSettingKey, false).toBool());
 
+  app_->SetCollectMemoryInfo(settings.value(kCollectMemoryInfoSettingKey, false).toBool());
+  uint64_t memory_sampling_period_ns = kMemorySamplingPeriodNsDefaultValue;
+  uint64_t memory_warning_threshold_kb = kMemoryWarningThresholdKbDefaultValue;
+  if (app_->GetCollectMemoryInfo()) {
+    memory_sampling_period_ns = settings
+                                    .value(kMemorySamplingPeriodNsSettingKey,
+                                           QVariant::fromValue(kMemorySamplingPeriodNsDefaultValue))
+                                    .toULongLong();
```

Comment:
Are we checking that nothing could crash if the user insert an invalid number here?

Reply:
Yep! We have the `UInt64Validator` to check the input number is valid or not.